File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/utils/PlayListHelper.java
package com.github.anrimian.musicplayer.domain.models.utils;

import com.github.anrimian.musicplayer.domain.models.composition.Composition;
import com.github.anrimian.musicplayer.domain.models.playlist.PlayList;
import com.github.anrimian.musicplayer.domain.utils.Objects;

import javax.annotation.Nonnull;

public class PlayListHelper {

    public static boolean hasChanges(@Nonnull PlayList first, @Nonnull PlayList second) {
        return !Objects.equals(first.getName(), second.getName())
                || !Objects.equals(first.getDateAdded(), second.getDateAdded())
                || !Objects.equals(first.getDateModified(), second.getDateModified())
                || first.getCompositionsCount() != second.getCompositionsCount()
                || first.getTotalDuration() != second.getTotalDuration();
    }
}

File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/utils/PlayListHelper.kt
package com.github.anrimian.musicplayer.domain.models.utils

import com.github.anrimian.musicplayer.domain.models.playlist.Playlist

object PlayListHelper {

    @JvmStatic
    fun areItemsTheSame(oldSource: Playlist, newSource: Playlist): Boolean {
        return oldSource.id == newSource.id
    }

    fun areSourcesTheSame(oldSource: Playlist, newSource: Playlist): Boolean {
        return !hasChanges(oldSource, newSource)
    }

    fun hasChanges(first: Playlist, second: Playlist): Boolean {
        return first.name != second.name
                || first.addedTime != second.addedTime
                || first.modifiedTime != second.modifiedTime
                || first.compositionsCount != second.compositionsCount
                || first.totalDuration != second.totalDuration
    }

}
